datalad_neuroimaging/bidsapp.py

```diff
@@ -0,0 +1,98 @@
+import logging
+from os.path import join as opj
+from datalad.interface.base import build_doc, Interface
+from datalad.support.param import Parameter
+from datalad.distribution.dataset import datasetmethod, EnsureDataset, \
+    require_dataset, resolve_path, Dataset
+from datalad.interface.utils import eval_results
+from datalad.support.constraints import EnsureStr
+from datalad.support.constraints import EnsureNone
+from datalad.support.exceptions import InsufficientArgumentsError
+from datalad.coreapi import run
+from datalad.interface.results import get_status_dict
+
+
+lgr = logging.getLogger("datalad.cbbsimaging.bidsapp")
+
+
+@build_doc
+class BidsApp(Interface):
+    """Run a BIDSApp on the dataset
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""dataset to run the BIDSApp on""",
+            constraints=EnsureDataset() | EnsureNone()),
+        cmd=Parameter(
+            args=("cmd",),
+            doc="""Preconfigured BIDSApp to run. The name corresponds to
+            datalad.neuroimaging.bidsapp.NAME configuration.""",
+            metavar="CMD",
+            constraints=EnsureStr() | EnsureNone())
+        )
```

Comment:
It would be very useful to expose all the aspects of preconfiguration also as arguments, so we can support ad-hoc use.

Reply:
Agree. I also assume, while making other BIDSApps available that way, we'll discover the need for slightly different configuration.